### PR TITLE
Dos2unix

### PR DESCRIPTION
**Pull request by henryyang42**

### CODE REVIEW

**Comment by nickelcheng** [comment]
+1, @CHIAHSINHOU please test

File: README.md [hunk 1]
```diff
@@ -40,18 +40,18 @@ git config --global url."https://".insteadOf git://
...
 * git
 * pillow
+* dos2unix
 
 ###Installation:
 * Install dependencies (for ubuntu)
 ```
 sudo apt-get install git
-sudo apt-get install mysql-server
+sudo apt-get install mysql-server python-mysqldb
```
Comment (nickelcheng) [inline README.md]: What's the point of putting them together？

Comment (henryyang42) [inline README.md]: The 2 are indispensable  and should be put together.

**Comment by nickelcheng** [comment]
This PR seems to be good. Unfortunately, no one has windows to test it :(

**Comment by henryyang42** [comment]
QAQ. So sad.

**Comment by nickelcheng** [comment]
2 problems
1. I can not download the testcase I just uploaded. (on 77.129, file ```11648.in``` and ```11648.out```)
2. no content in the file

**Comment by henryyang42** [comment]
I can't download testcase in current ``dev`` too.

**Comment by nickelcheng** [comment]
@geniusgordon can you check what's wrong?

**Comment by geniusgordon** [comment]
#217

**Comment by henryyang42** [comment]
no content in the file issue fixed

**Comment by nickelcheng** [comment]
That's because upload failed.
I'll test again.

File: problem/views.py [hunk 2]
```diff
@@ -232,14 +232,18 @@ def testcase(request, pid, tid=None):
...
-                with open('%s%s.in' % (TESTCASE_PATH, testcase.pk), 'w') as t_in:
+                input_filename = '%s%s.in' % (TESTCASE_PATH, testcase.pk)
+                output_filename = '%s%s.out' % (TESTCASE_PATH, testcase.pk)
+                with open(input_filename, 'w') as t_in:
                     for chunk in request.FILES['t_in'].chunks():
-                        t_in.write(chunk.replace('\r\n', '\n'))
-                    logger.info("testcase %s.in saved by %s" % (testcase.pk, request.user))
-                with open('%s%s.out' % (TESTCASE_PATH, testcase.pk), 'w') as t_out:
+                        t_in.write(chunk)
+                call(['dos2unix', input_filename])
```
Comment (bruce3557) [inline problem/views.py]: prefer to use ```check_call``` instend of call.
but you can customize the response of result with ```Popen``` function

**Comment by henryyang42** [comment]
Change to ``check_call`` now

**Comment by nickelcheng** [comment]
test ok, @bruce3557 do you have any other suggestion?

File: problem/views.py [hunk 3]
```diff
@@ -232,14 +232,18 @@ def testcase(request, pid, tid=None):
...
-                with open('%s%s.in' % (TESTCASE_PATH, testcase.pk), 'w') as t_in:
+                input_filename = '%s%s.in' % (TESTCASE_PATH, testcase.pk)
+                output_filename = '%s%s.out' % (TESTCASE_PATH, testcase.pk)
+                with open(input_filename, 'w') as t_in:
                     for chunk in request.FILES['t_in'].chunks():
-                        t_in.write(chunk.replace('\r\n', '\n'))
-                    logger.info("testcase %s.in saved by %s" % (testcase.pk, request.user))
-                with open('%s%s.out' % (TESTCASE_PATH, testcase.pk), 'w') as t_out:
+                        t_in.write(chunk)
+                check_call(['dos2unix', input_filename])
```
Comment (bruce3557) [inline problem/views.py]: Sorry but I need further checking.
Could you please check if ```check_call``` throws an exception to see whether system is crashed?
Thanks!

**Comment by bruce3557** [comment]
If the guessing is checked, I have no problem with this

**Comment by bruce3557** [comment]
going to merge

**Pull request merged**